utils/build_ds_container.py

```diff
@@ -0,0 +1,257 @@
+        'that you properly tag built images for production use. '
+        'Auto-generated tags are primarily useful for development workflows. '
+        'This script assumes you have authenticated to the image repository '
+        'if necessary (e.g, `podman login`).'))
+parser.add_argument(
+    '-t', '--container-image-tag',
+    help='A unique tag for the container image.')
+args = parser.parse_args()
+
+
+log.basicConfig(
+    format='%(asctime)s:%(levelname)s: %(message)s', level=log.INFO)
+
+
+# FIXME(lbragstad): Remove this and replace it with operating system utils if
+# possible.
+result = subprocess.run(
+    ['git', 'rev-parse', '--show-toplevel'], capture_output=True, check=True)
+# Convert the file path from bytes to unicode since we might manipulate it
+# later. Also, strip off any newlines.
+REPO_PATH = result.stdout.decode().strip()
+
+
+def ensure_namespace_exists(namespace):
+    """Function that ensures there is a namespace for the content."""
+    with client.ApiClient() as api_client:
+        body = client.V1Namespace(
+            api_version='v1', kind='Namespace', metadata={'name': namespace})
+
+        try:
+            api = client.CoreV1Api(api_client)
+            api.create_namespace(body)
+            log.info('Created namespace %s', namespace)
+        except exceptions.ApiException as err:
+            if err.status == 409:
+                log.info('Reusing existing namespace: %s', namespace)
+            else:
+                raise err
+
+
+def supports_custom_resource(resource_name):
+    with client.ApiClient() as api_client:
+        api = client.ApiextensionsV1Api(api_client)
+        try:
+            api.read_custom_resource_definition(resource_name)
+            return True
+        except exceptions.ApiException as err:
+            return False
+            if err.status == 404:
+                return False
+            log.warning(
+                'Unable to determine if %s custom resources are supported in '
+                'this deployment', resource_name)
+            return False
+
+
+def create_image_stream():
+    # FIXME(lbragstad) Fill this out using an openshift client, since we're
+    # dealing with an openshift-specific resource.
+    with client.ApiClient() as api_client:
+        body = client.V1CustomResourceDefinition(
+            api_version='image.openshift.io/v1',
+            kind='ImageStream',
+            metadata={'name': 'openscap-ocp4-ds'},
+            spec={'lookupPolicy': {'local': True}})
+
+        api = client.ApiextensionsV1Api(api_client)
+        api.create_custom_resource_definition(body)
+
+
+def create_build_config(build_config):
+    # FIXME(lbragstad) Fill this out using an openshift client, since we're
+    # dealing with an openshift-specific resource.
+    pass
+
+
+def build_content_locally(products, debug=False):
+    build_binary_path = os.path.join(REPO_PATH, 'build_product')
+    command = [build_binary_path] + products
```

Comment:
Ok, I see, this is an array

Reply:
Good question - it can be done either way in python. The python community issued guidance via a PEP that recommends using whitespace around operators.

https://www.python.org/dev/peps/pep-0008/#other-recommendations